lib/database/repo.ex

```diff
@@ -1,48 +1,55 @@
-defmodule BorsNG.Database.Repo do
+defmodule DynamicEctoRepoWrapper do
   @moduledoc """
-  An ecto data repository;
-  this process interacts with your persistent database.
-
-  Do not confuse this with a GitHub repo.
-  We call those `Project`s internally.
+  Exposes a macro to define Ecto.Repo functions dynamically.
   """
+  defmacro create_ecto_repo_callback(args, name) do
+    quote bind_quoted: [args: args, name: name] do
+      def unquote(name)(unquote_splicing(args)) do
+        repo = :persistent_term.get(:db_repo)
+        apply(repo, unquote(name), unquote(args))
+      end
+    end
+  end
 
-  use Ecto.Repo,
-    otp_app: :bors,
-    adapter: Application.get_env(:bors, BorsNG.Database.Repo)[:adapter]
+  def create_ecto_repo_callback_args(_, 0) do
+    []
+  end
 
-  def init(_, config) do
-    # Backwards compatibility hack: if POSTGRES_HOST is set, and the database URL is left at default,
-    # use the older configuration.
-    config = Confex.Resolver.resolve!(config)
-    no_host = is_nil(System.get_env("POSTGRES_HOST"))
+  def create_ecto_repo_callback_args(module, arity) do
+    Enum.map(1..arity, &Macro.var(:"arg#{&1}", module))
```

Comment:
Any way to define default arguments here, sparing us from having to define the functions explicitly with multiple arities?